File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int n = nums.size();
        int maxi = 0;
        for(int i=0;i<n;i++)
        {
            maxi = max(maxi, nums[i]);
        }

        int cnt = 0;
        int maxim = 0;
        for(int i=0;i<n;i++)
        {
            if(nums[i] == maxi)  {
                cnt++;
                maxim = max(cnt,maxim);
            }
            else {
                cnt = 0;
            }
        }

        return maxim;
    }
};

File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.java
class Solution {
    public int longestSubarray(int[] nums) {
        int n = nums.length;
        int maxi = 0;
        int cnt = 0;
        int ans = 0;
        for(int i=0;i<n;i++)
        {
            maxi = Math.max(nums[i], maxi);
        }

        for(int i=0;i<n;i++)
        {
            if(nums[i] == maxi)
            {
                cnt++;
                ans = Math.max(ans, cnt);
            }
            else {
                cnt = 0;
            }
        }

        return ans;
    }
}